### ISSUE TITLE
crashed using latest git version:

### ISSUE THREAD

**Issue by thestick613** [ElementsProject/lightning#577]
```
bitcoin-cli -testnet -datadir=/data/bitcoin-data-testnet/ gettxout 318ed3bdc7bdbdc7db07ff7cb9021d04f841abae4403298f80e9440a77497d01 1: had bad value ({
  "bestblock": "00000000000001e3f557f71fe043f578d5c7aa7fa2440cea2d397d78666d8f42",
  "confirmations": 62181,
  "value": 0.00800000,
  "scriptPubKey": {
    "asm": "0 558068a61b5569b9f5991ad8c674f5b8ad40125df81faa7d47d8df0c725514b4",
    "hex": "0020558068a61b5569b9f5991ad8c674f5b8ad40125df81faa7d47d8df0c725514b4",
    "type": "witness_v0_scripthash"
  },
  "coinbase": false
} 
)?
lightningd(25614): bitcoin-cli -testnet -datadir=/data/bitcoin-data-testnet/ gettxout 318ed3bdc7bdbdc7db07ff7cb9021d04f841abae4403298f80e9440a77497d01 1: had bad value ({
  "bestblock": "00000000000001e3f557f71fe043f578d5c7aa7fa2440cea2d397d78666d8f42",
  "confirmations": 62181,
  "value": 0.00800000,
  "scriptPubKey": {
    "asm": "0 558068a61b5569b9f5991ad8c674f5b8ad40125df81faa7d47d8df0c725514b4",
    "hex": "0020558068a61b5569b9f5991ad8c674f5b8ad40125df81faa7d47d8df0c725514b4",
    "type": "witness_v0_scripthash"
  },
  "coinbase": false
} 
)?
Log dumped in crash.log
lightningd(25614): FATAL SIGNAL 6 RECEIVED
lightningd(25614): backtrace: lightningd/log.c:436 (log_crash) 0x4107ef
lightningd(25614): backtrace: (null):0 ((null)) 0x2ba070dc04af
lightningd(25614): backtrace: (null):0 ((null)) 0x2ba070dc0428
lightningd(25614): backtrace: (null):0 ((null)) 0x2ba070dc2029
lightningd(25614): backtrace: lightningd/log.c:525 (fatal) 0x410c29
lightningd(25614): backtrace: lightningd/bitcoind.c:458 (process_gettxout) 0x4049fa
lightningd(25614): backtrace: lightningd/bitcoind.c:157 (bcli_finished) 0x403c0b
lightningd(25614): backtrace: ccan/ccan/io/poll.c:183 (destroy_conn) 0x458498
lightningd(25614): backtrace: ccan/ccan/io/poll.c:189 (destroy_conn_close_fd) 0x4584b8
lightningd(25614): backtrace: ccan/ccan/tal/tal.c:240 (notify) 0x461b43
lightningd(25614): backtrace: ccan/ccan/tal/tal.c:400 (del_tree) 0x461ff4
lightningd(25614): backtrace: ccan/ccan/tal/tal.c:509 (tal_free) 0x462407
lightningd(25614): backtrace: ccan/ccan/io/io.c:443 (io_close) 0x457120
lightningd(25614): backtrace: ccan/ccan/io/poll.c:309 (io_loop) 0x458896
lightningd(25614): backtrace: lightningd/lightningd.c:350 (main) 0x40f234
lightningd(25614): backtrace: (null):0 ((null)) 0x2ba070dab82f
lightningd(25614): backtrace: (null):0 ((null)) 0x403138
lightningd(25614): backtrace: (null):0 ((null)) 0xffffffffffffffff
Fatal signal 6. 0x4108d5 log_crash
        lightningd/log.c:462
0x2ba070dc04af ???
        ???:0
0x2ba070dc0428 ???
        ???:0
0x2ba070dc2029 ???
        ???:0
0x410c29 fatal
        lightningd/log.c:525
0x4049fa process_gettxout
        lightningd/bitcoind.c:458 
0x403c0b bcli_finished
        lightningd/bitcoind.c:157 
0x458498 destroy_conn
        ccan/ccan/io/poll.c:183   
0x4584b8 destroy_conn_close_fd
        ccan/ccan/io/poll.c:189   
0x461b43 notify
        ccan/ccan/tal/tal.c:240   
0x461ff4 del_tree
        ccan/ccan/tal/tal.c:400   
0x462407 tal_free
        ccan/ccan/tal/tal.c:509   
0x457120 io_close
        ccan/ccan/io/io.c:443
0x458896 io_loop
        ccan/ccan/io/poll.c:309   
0x40f234 main
        lightningd/lightningd.c:350
0x2ba070dab82f ???
        ???:0
0x403138 ???
        ???:0
0xffffffffffffffff ???
        ???:0
Log dumped in crash.log
Aborted (core dumped)
```

**Comment by bjd** [comment]
Same here, on a different txout but otherwise exact same stacktrace.

```
2018-01-12 12:22:39.202510500 lightningd(18489): bitcoin-cli -testnet -datadir=/storage/testnet gettxout 913c1225a31572ea76a02e93d85e4ed5019631cba304c5893467cae094776852 0: had bad value ({
2018-01-12 12:22:39.202512500   "bestblock": "00000000000001e3f557f71fe043f578d5c7aa7fa2440cea2d397d78666d8f42",
2018-01-12 12:22:39.202512500   "confirmations": 62197,
2018-01-12 12:22:39.202513500   "value": 0.02745568,
2018-01-12 12:22:39.202513500   "scriptPubKey": {
2018-01-12 12:22:39.202513500     "asm": "0 966760ce0d08d9a333a22c6b40544fe9679ddad19333e5f96c7789e244da0a1a",
2018-01-12 12:22:39.202514500     "hex": "0020966760ce0d08d9a333a22c6b40544fe9679ddad19333e5f96c7789e244da0a1a",
2018-01-12 12:22:39.202515500     "type": "witness_v0_scripthash"
2018-01-12 12:22:39.202515500   },
2018-01-12 12:22:39.202515500   "coinbase": false
2018-01-12 12:22:39.202515500 }
2018-01-12 12:22:39.202516500 )?
2018-01-12 12:22:39.203205500 lightningd(18489): FATAL SIGNAL 6 RECEIVED
2018-01-12 12:22:39.207956500 lightningd(18489): backtrace: lightningd/log.c:436 (log_crash) 0x4107ef
2018-01-12 12:22:39.207967500 lightningd(18489): backtrace: (null):0 ((null)) 0x7ff724e194af
2018-01-12 12:22:39.207973500 lightningd(18489): backtrace: (null):0 ((null)) 0x7ff724e19428
2018-01-12 12:22:39.207978500 lightningd(18489): backtrace: (null):0 ((null)) 0x7ff724e1b029
2018-01-12 12:22:39.207984500 lightningd(18489): backtrace: lightningd/log.c:525 (fatal) 0x410c29
2018-01-12 12:22:39.208145500 lightningd(18489): backtrace: lightningd/bitcoind.c:458 (process_gettxout) 0x4049fa
2018-01-12 12:22:39.208149500 lightningd(18489): backtrace: lightningd/bitcoind.c:157 (bcli_finished) 0x403c0b
2018-01-12 12:22:39.208240500 lightningd(18489): backtrace: ccan/ccan/io/poll.c:183 (destroy_conn) 0x458498
2018-01-12 12:22:39.208247500 lightningd(18489): backtrace: ccan/ccan/io/poll.c:189 (destroy_conn_close_fd) 0x4584b8
2018-01-12 12:22:39.208401500 lightningd(18489): backtrace: ccan/ccan/tal/tal.c:240 (notify) 0x461b43
2018-01-12 12:22:39.208409500 lightningd(18489): backtrace: ccan/ccan/tal/tal.c:400 (del_tree) 0x461ff4
2018-01-12 12:22:39.208415500 lightningd(18489): backtrace: ccan/ccan/tal/tal.c:509 (tal_free) 0x462407
2018-01-12 12:22:39.208515500 lightningd(18489): backtrace: ccan/ccan/io/io.c:443 (io_close) 0x457120
2018-01-12 12:22:39.208521500 lightningd(18489): backtrace: ccan/ccan/io/poll.c:309 (io_loop) 0x458896
2018-01-12 12:22:39.208632500 lightningd(18489): backtrace: lightningd/lightningd.c:350 (main) 0x40f234
2018-01-12 12:22:39.208638500 lightningd(18489): backtrace: (null):0 ((null)) 0x7ff724e0482f
2018-01-12 12:22:39.208643500 lightningd(18489): backtrace: (null):0 ((null)) 0x403138
2018-01-12 12:22:39.208647500 lightningd(18489): backtrace: (null):0 ((null)) 0xffffffffffffffff
```

**Comment by asocjal** [comment]
Also crashes on my environment:

> +68520.396566480 lightningd(1898):UNUSUAL: bitcoin-cli -testnet estimatesmartfee 2 CONSERVATIVE exited with status 1
+68520.400690987 lightningd(1898):BROKEN: bitcoin-cli -testnet estimatesmartfee 2 CONSERVATIVE: invalid response
+68529.500843283 lightningd(1898):BROKEN: FATAL SIGNAL 6 RECEIVED
+68529.518318037 lightningd(1898):BROKEN: backtrace: lightningd/log.c:436 (log_crash) 0x40fcb6
+68529.521383265 lightningd(1898):BROKEN: backtrace: (null):0 ((null)) 0x7ff25daff4af
+68529.521968794 lightningd(1898):BROKEN: backtrace: (null):0 ((null)) 0x7ff25daff428
+68529.522514177 lightningd(1898):BROKEN: backtrace: (null):0 ((null)) 0x7ff25db01029
+68529.522562406 lightningd(1898):BROKEN: backtrace: lightningd/log.c:525 (fatal) 0x4100f0
+68529.522724451 lightningd(1898):BROKEN: backtrace: lightningd/bitcoind.c:249 (extract_feerate) 0x40404a
+68529.522776608 lightningd(1898):BROKEN: backtrace: lightningd/bitcoind.c:285 (process_estimatefee) 0x404103
+68529.522823686 lightningd(1898):BROKEN: backtrace: lightningd/bitcoind.c:156 (bcli_finished) 0x403c0b
+68529.522963066 lightningd(1898):BROKEN: backtrace: ccan/ccan/io/poll.c:183 (destroy_conn) 0x457463
+68529.523019011 lightningd(1898):BROKEN: backtrace: ccan/ccan/io/poll.c:189 (destroy_conn_close_fd) 0x457483
+68529.523194031 lightningd(1898):BROKEN: backtrace: ccan/ccan/tal/tal.c:240 (notify) 0x460b0e
+68529.523244160 lightningd(1898):BROKEN: backtrace: ccan/ccan/tal/tal.c:400 (del_tree) 0x460fbf
+68529.523291337 lightningd(1898):BROKEN: backtrace: ccan/ccan/tal/tal.c:509 (tal_free) 0x4613d2
+68529.523437334 lightningd(1898):BROKEN: backtrace: ccan/ccan/io/io.c:443 (io_close) 0x4560eb
+68529.523490609 lightningd(1898):BROKEN: backtrace: ccan/ccan/io/poll.c:309 (io_loop) 0x457861
+68529.523632379 lightningd(1898):BROKEN: backtrace: lightningd/lightningd.c:354 (main) 0x40e6fb
+68529.523681611 lightningd(1898):BROKEN: backtrace: (null):0 ((null)) 0x7ff25daea82f
+68529.524121745 lightningd(1898):BROKEN: backtrace: (null):0 ((null)) 0x403138
+68529.524166890 lightningd(1898):BROKEN: backtrace: (null):0 ((null)) 0xffffffffffffffff

> +28.248049450 lightningd(27442):BROKEN: bitcoin-cli -testnet gettxout 318ed3bdc7bdbdc7db07ff7cb9021d04f841abae4403298f80e9440a77497d01 1: had bad value ({
  "bestblock": "000000000000015909fe48f2cb27d0b1e4a4e5cbd2eaf61e177a53437e185bba",
  "confirmations": 62192,
  "value": 0.00800000,
  "scriptPubKey": {
    "asm": "0 558068a61b5569b9f5991ad8c674f5b8ad40125df81faa7d47d8df0c725514b4",
    "hex": "0020558068a61b5569b9f5991ad8c674f5b8ad40125df81faa7d47d8df0c725514b4",
    "type": "witness_v0_scripthash"
  },
  "coinbase": false
}
)?
+28.483753628 lightningd(27442):BROKEN: FATAL SIGNAL 6 RECEIVED
+28.504404356 lightningd(27442):BROKEN: backtrace: lightningd/log.c:436 (log_crash) 0x4107ef
+28.511299946 lightningd(27442):BROKEN: backtrace: (null):0 ((null)) 0x7f270e9c64af
+28.511430577 lightningd(27442):BROKEN: backtrace: (null):0 ((null)) 0x7f270e9c6428
+28.511509859 lightningd(27442):BROKEN: backtrace: (null):0 ((null)) 0x7f270e9c8029
+28.511583746 lightningd(27442):BROKEN: backtrace: lightningd/log.c:525 (fatal) 0x410c29
+28.511891473 lightningd(27442):BROKEN: backtrace: lightningd/bitcoind.c:458 (process_gettxout) 0x4049fa
+28.511972942 lightningd(27442):BROKEN: backtrace: lightningd/bitcoind.c:157 (bcli_finished) 0x403c0b
+28.512347715 lightningd(27442):BROKEN: backtrace: ccan/ccan/io/poll.c:183 (destroy_conn) 0x458498
+28.512441219 lightningd(27442):BROKEN: backtrace: ccan/ccan/io/poll.c:189 (destroy_conn_close_fd) 0x4584b8
+28.512720613 lightningd(27442):BROKEN: backtrace: ccan/ccan/tal/tal.c:240 (notify) 0x461b43
+28.512797329 lightningd(27442):BROKEN: backtrace: ccan/ccan/tal/tal.c:400 (del_tree) 0x461ff4
+28.512859906 lightningd(27442):BROKEN: backtrace: ccan/ccan/tal/tal.c:509 (tal_free) 0x462407
+28.513036669 lightningd(27442):BROKEN: backtrace: ccan/ccan/io/io.c:443 (io_close) 0x457120
+28.513143375 lightningd(27442):BROKEN: backtrace: ccan/ccan/io/poll.c:309 (io_loop) 0x458896
+28.513359521 lightningd(27442):BROKEN: backtrace: lightningd/lightningd.c:350 (main) 0x40f234
+28.513441038 lightningd(27442):BROKEN: backtrace: (null):0 ((null)) 0x7f270e9b182f
+28.513515003 lightningd(27442):BROKEN: backtrace: (null):0 ((null)) 0x403138
+28.513589443 lightningd(27442):BROKEN: backtrace: (null):0 ((null)) 0xffffffffffffffff

**Comment by cdecker** [comment]
I guess this is fixed by #578, can you confirm this @thestick613?

**Comment by thestick613** [comment]
Been running for 5 minutes without a crash, so it looks okay now!
